### PR TITLE
Issue 600 Add doc with instructions on how to create a release

### LINKED ISSUE #600: Add instructions on how to build releases to readme
Like we do at https://github.com/opendatakit/javarosa#publishing-the-jar-to-ossrh-and-maven-central, we should have a section called `Publishing the jar` that roughly describes how we build and release the jar.

I think the important, non-obvious things are...
1. You need to have a tag so the version is correct
1. You need to have the logback and gradle properties
1. We build with Java 1.8
1. You gotta run some magic command to tag the Sentry release

We can probably leave out things like writing to the forum.

### PR DESCRIPTION
**Pull request by ggalmazor**
Closes #600

This PR adds a doc with instructions on how to release a new version of Briefcase

### CODE REVIEW

**Comment by ggalmazor** [comment]
- I still need to add info about:
  - the log file and GA tracking code used in production (without revealing any sensible data)
  - sending refs to Sentry to link the release commits

**Comment by ggalmazor** [comment]
I'm happy with the current contents. I think they're comprehensive and I think (hope) I haven't missed anything important.

We can add instructions for patch versions when we release the next patch version.

File: docs/how-to-release.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+   
+   - Select the `ODK Briefcase` app
+   - Go to `Admin > (Property section) Property Settings`
+   
+   Tracking IDs follow the pattern `UA-12345678-1`
+     
+  
+7. Build the release JAR file with `./gradlew clean build -xtest`
+
+    The JAR file will be located at `build/libs` and the filename should be like: `ODK Briefcase vx.y.z.jar`. (replace x.y.z with the actual numbers you've used in step 2)
```
Comment (yanokwa) [inline docs/how-to-release.md]: Maybe our Gradle task should just do that and we can drop the text under 8.

File: docs/how-to-release.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
+# How to release a new Briefcase version
```
Comment (yanokwa) [inline docs/how-to-release.md]: s/new//g since we don't release old versions 😂

**Comment by yanokwa** [comment]
Might also be time to send in an README.md for the docs folder.

File: docs/how-to-release.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+      * [Releasing a minor version](#releasing-a-minor-version)
+         * [Releasing a beta of a minor version](#releasing-a-beta-of-a-minor-version)
+      * [Releasing a hot fix (patch version)](#releasing-a-hot-fix-patch-version)
+      * [Post-release considerations](#post-release-considerations)
+
+
+## About semantic versioning
+
+- Briefcase uses semantic versioning: vx.y.z
+  - x represents the *mayor* version number. Different mayor versions can be incompatible between them
```
[the remark this answers is not in the document]
Comment (ggalmazor) [reply docs/how-to-release.md]: Ugh, thanks!

File: docs/how-to-release.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  - y represents the *minor* version number. Different minor versions of the same mayor versions are compatible between them and bring new features
+  - z represents the *patch* version number. Patch versions are use to release bug fixes only.
+
+## Releasing a minor version
```
[the remark this answers is not in the document]
Comment (ggalmazor) [reply docs/how-to-release.md]: I think we should leave `minor` everywhere to differentiate the instructions (yet to be added to this doc) for releasing a patch version, which have some important nuances.

**Comment by ggalmazor** [comment]
I've updated the PR, @yanokwa. There are still some topics to discuss:
- All the `minor` mentions, which I'd prefer to keep since we will have to explain how to release patch versions too.
- The JAR renaming instructions
- Review of the author examples on the release notes template

File: docs/how-to-release.md (hunk 1 above)
Comment (ggalmazor) [reply docs/how-to-release.md]: I went ahead and changed the build file to fix this

File: docs/how-to-release.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+- The process is basically the same as with a minor version with some small differences:
+  - We suffix the version number with `-beta.N`, replacing `N` with the number of the beta, starting with `0`
+  - We check the `This is a pre-release` checkbox to avoid going to it with the link https://github.com/opendatakit/briefcase/releases/latest
+  
+## Post-release considerations
+
+- The new release should be linked in Sentry by sending the git refs involved (usually, the last from previous release and the last from the new release).
+
+  The following curl carl with take care of this:
```
Comment (yanokwa) [inline docs/how-to-release.md]: s/carl/command/

File: docs/how-to-release.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    You should see a message telling you about the new tag GitHub has created for your release
+
+5. Check out the tag with `git checkout vx.y.z`. (replace x.y.z with the actual numbers you've used in step 2)
+5. Check out the tag with `git checkout vx.y.z`. (replace x.y.z with the actual numbers you've used in step 2)
+
+    Git should announce that you're on a `detached HEAD`. That's OK.
+    
+6. Verify that logging, error monitoring, and user tracking have production configurations
+
+   You typically have to do this once if you keep a separate release workspace.
```
Comment (yanokwa) [inline docs/how-to-release.md]: s/typically/only/

File: docs/how-to-release.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+   
+   - Select the `ODK Briefcase` app
+   - Go to `Admin > (Property section) Property Settings`
+   
+   Tracking IDs follow the pattern `UA-12345678-1`
+     
+  
+7. Build the release JAR file with `./gradlew clean build`
+
+    The JAR file will be located at `build/libs` and the filename should be like: `ODK Briefcase vx.y.z.jar`. (replace x.y.z with the actual numbers you've used in step 2)
```
Comment (yanokwa) [inline docs/how-to-release.md]: It's now `ODK-Briefcase-vx.y.z.jar`

File: docs/how-to-release.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+  
+    If you JAR's filename has `-dirty` or any other deviation from the previous template, you've made some mistake while sync'ing your repo after publishing the release on GitHub.
```
Comment (yanokwa) [inline docs/how-to-release.md]: s/If you/If your/
s/sync'ing/syncing/

File: test/resources/org/opendatakit/briefcase/export/simple-form-dupe-fields.txt [hunk 9]
```diff
@@ -0,0 +1,53 @@
+<!--
```
Comment (yanokwa) [inline test/resources/org/opendatakit/briefcase/export/simple-form-dupe-fields.txt]: This looks totally unrelated.

**Comment by yanokwa** [comment]
I want to push back on the `minor` issue. Agreed that you have to explain how to release patch versions, but you do that already in the text so I'm not sure why that's relevant.

Currently, we don't have a process for a major release. It's likely to be the same as a minor release, so my brain prefers us just to have language around "releases" and "hot fixes/patches". I guess when it's written the way it is, it feels like we forgot to write the section on major releases.

**Comment by ggalmazor** [comment]
> At some point, we should have a conversation about what it would take to go to 2.0.

Regarding new features we're planning, we know that the ISO8601 adaptation will break backwards compatibility

From an architectural point of view, there are two big changes we want to make: 
- The centralized database
- New concurrency model

Other external factors that we should probably take into account:
- We should start thinking about Java 11, which is the new LTS version

If feels like there are a lot of waves converging to the 2.0 release.

We could also consider more marketable version names, like "Briefcase Returns" or "Briefcase II: The standard strikes back" :D

File: build.gradle [hunk 10]
```diff
@@ -137,7 +137,7 @@ mainClassName = 'org.opendatakit.briefcase.Launcher'
 jar {
     baseName = buildConfig.appName
     version = buildConfig.version
-    archiveName = baseName + ' ' + version + '.jar'
+    archiveName = 'ODK-Briefcase-' + version + '.jar'
```
Comment (lognaturel) [inline build.gradle]: Was this change discussed somewhere? I imagine it will break some downstream tools.

Comment (lognaturel) [reply build.gradle]: Oh man. @yanokwa just explained it to me. The releases on GitHub have dashes because spaces are replaced. This makes local builds match.

Ok, I approve. Weird.

Comment (ggalmazor) [reply build.gradle]: It was "discussed" here: https://github.com/opendatakit/briefcase/pull/634#discussion_r219316223
